### DISCUSSION TITLE
Two dialogue events got mixed up

### DISCUSSION THREAD

**Discussion by ilmarilahti** [FakeFishGames/Barotrauma#14608] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I entered an outpost in a single player campaign in the first biome. The "teens come asking you questions" event was triggered, and then in the middle of its dialogue sequence the "radio blaring nearby" event was also triggered and seemed to take over the dialogue UI, so that once I had clicked through that one, the dialogue went back to the last part of the teens event sequence.

### Reproduction steps

Play long enough to trigger two events simultaneously, or, probably better, somehow simulate that situation. I'm not sure if this could be something specific to just these two events ("teens" and "radio") and whether those are available after the first biome.

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

Other

### -

1.6.0.0

### Which operating system did you encounter this bug on?

MacOS

**Comment by ubertpendragon** [comment]
That's just what happens when events use timed/proximity triggers, you can be in the middle of a dialogue and then you get dialogue for another event

The events don't queue, they all technically share the same dialogue box

**Reply by Regalis11** [reply]
> Modder explaining one of the programmers how the game works technically
 
😉 /jk

This is not true, they do queue in many situations (see `ConversationAction.IsBlockedByAnotherConversation`). They also do not all share the same dialogue box.

I'm not sure why `ConversationAction.IsBlockedByAnotherConversation` doesn't work in this case though, but on a cursory look through the code, I think it might have to do with one of the conversations being a "global" one that's shown to all players (instead of a specific target).

**Reply by ubertpendragon** [reply]
Oh alright that makes sense, I have a few events that open a conversation to "player" when something bad happens and that's where the dialogue tends to insert itself within another

**Reply by ubertpendragon** [reply]
Just do any event and then `triggerevent bigbrother`

![image](https://github.com/user-attachments/assets/aafd2a70-cdbb-4465-ac6c-2e8206a9889c)

**Comment by Regalis11** [comment]
This has been fixed in our private repo now, closing. Unfortunately the fix didn't make it to the next update (the PvP update), but will be coming in the one after that.
